src/dataflow/src/sink/tail.rs

```diff
@@ -56,18 +68,26 @@ pub fn tail<G>(
                     while cursor.val_valid(&batch) {
                         let row = cursor.val(&batch);
                         cursor.map_times(&batch, |time, diff| {
-                            assert!(*diff >= 0, "negative multiplicities sinked in tail");
```

Comment:
Did removing the `consolidate` call get rid of the chance of negative multiplicities?

Reply:
Even before, there was no chance of negative multiplicities. Except maybe with CDCv2 sources? 🤔 I'll try that.

But it's probably best not to remove the assert, I'll undo that.

Reply:
Scratch that, I now remembered the real reason. I'm adding this as a comment on the commit:

Expanding a bit, the unrolled pipeline for TAIL, combining the
"envelope" and the sink code, was roughly as shown below. I'm taking
some artistic licence here, using Streams where the actual code as
Collections, to show the types more clearly, and other shortcuts:

```
let input: Stream<(Row, time, diff)> = ...
let enriched: Stream<(Row, time, 1)> = input
  .consolidate()
  .map(|(row, time, diff)| {
    let wrapping_row = Row::new();
    if (emit_progress) {
      wrapping_row.push(false);
    }
    wrapping_row.push(time);
    wrapping_row.push(diff);
    wrapping_row.push(row);
    (<some key>, (wrapping_row, time, 1))
  }
  .arrange_by_key();

sink::tail(enriched);
```

So the actual rows, along with their time/diff where put into wrapping
rows. And tail() would write out those rows.

Now, we lose the first consolidate() call, have one arrange_by_key(),
and then write out the user/vanilla rows straight from that, and create
the wrapping Row right in the sink:tail() code, without sending it along
Streams.

The assert! that we removed in sink::tail() was asserting on the diff of
the wrapping row. Now we straight get the vanilla rows, which can (and
should) have negative multiplicities.

Reply:
Awesome, that makes sense!